src/js/controllers/index.js

```diff
@@ -693,29 +649,22 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     var step = 6;
 
     var unique = {};
+
     function getHistory(skip, cb) {
-      skip = skip || 0;
-      fc.getTxHistory({
-        skip: skip,
-        limit: step,
-      }, function(err, txs) {
+      storageService.getTxHistory(c.walletId, function(err, txs) {
         if (err) return cb(err);
-        if (txs && txs.length > 0) {
-          lodash.each(txs, function(tx) {
-            if (!unique[tx.txid]){
-              allTxs.push(tx);
-              unique[tx.txid] = 1;
-              console.log("Got:" + lodash.keys(unique).length + " txs");
-            } else {
-              console.log("Ignoring duplicate TX in CSV: "+ tx.txid);
-            }
-          });
-          return getHistory(skip + step, cb);
-        } else {
-          return cb(null, lodash.flatten(allTxs));
+
+        var txsFromLocal;
+        try {
+          txsFromLocal = JSON.parse(txs);
+        } catch (ex) {
+          return cb(ex);
         }
+
+        allTxs.push(txsFromLocal);
+        return cb(null, lodash.flatten(allTxs));
       });
-    };
+    }
 
     if (isCordova) {
```

Comment:
This check should not be performed, this function should never be called if `!isCordoba`.

Or it should throw:
if (isCordoba) throw ('CSV generation not available in mobile');